### PR TITLE
[CODEOWNERS] Add code owners for DPC++ tools

### PR DESCRIPTION
**Pull request by bader**
Assign code owners for file-table-tform, llvm-foreach, llvm-no-spir-kernel tools and ESIMD implementation.

Add more comments and co-locate related components.

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -78,16 +90,16 @@ sycl/source/half_type.cpp @AlexeySachkov
...
-
-sycl/doc/extensions/ @intel/dpcpp-specification-reviewers
-
+# XPTI instrumentation utilities
 xpti/ @tovinkere @andykaylor
 xptifw/ @tovinkere  @andykaylor
 
+# DPC++ tools
+llvm/tools/file-table-tform/ @kbobrovs @AlexeySachkov
+llvm/tools/llvm-foreach/ @Fznamznon @AlexeySachkov
```
Comment (Fznamznon) [inline .github/CODEOWNERS:99]: I suggest @AlexeySachkov as primary code owner for this tool.
```suggestion
llvm/tools/llvm-foreach/ @AlexeySachkov  @Fznamznon
```

**Review by Fznamznon — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,18 +1,27 @@
 * @bader
 
+# Front-ent compiler
```
Comment (AGindinson) [inline .github/CODEOWNERS:3]:
```suggestion
# Front-end compiler
```

**Review by AGindinson — DISMISSED** [review]

**Comment by romanovvlad** [comment]
@kbobrovs @AlexeySachkov @Fznamznon @AGindinson 
Could you please [re]approve?

**Review by Fznamznon — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Review by AGindinson — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]
I approve likes where I'm the owner. One note:
> llvm/tools/llvm-no-spir-kernel/ @AGindinson @AlexeySachkov

we may want to include fpga folks here as well

**Pull request merged**